downloader/src/main/kotlin/vcs/Git.kt

```diff
@@ -111,7 +113,7 @@ object Git : GitBase() {
 
             val workingTree = getWorkingTree(targetDir)
 
-            val revision = if (vcs.revision.isNotBlank()) {
+            val revision = if (vcs.revision.isNotBlank() && vcs.revision !in movingRevisionNames) {
```

Comment:
Repeating this condition in all VCS seems error prone, why not make a function in the parent class, something like `revisionIsValid`?

Reply:
Done.